alerter-component/src/main/java/io/github/steliospaps/ighackathon/instrumentpricealerter/alertercomponent/websocket/client/WebsocketClient.java

```diff
@@ -122,6 +121,7 @@ public void shutdown() {
 				.log("ws-output",Level.FINEST,SignalType.ON_NEXT)//
 				.log("ws-output",Level.INFO,SignalType.ON_SUBSCRIBE,SignalType.CANCEL,SignalType.ON_COMPLETE,SignalType.ON_ERROR,SignalType.AFTER_TERMINATE)//
 				.map(toSend->ws.textMessage(toSend)))
+				.doOnTerminate(() -> log.warn("will retry connection in {} seconds", retryConnectionInterval.getSeconds()))
```

Comment:
doOnTerminate does not hande cancel consider:
doFinally(Consumer<SignalType> onFinally)
Add behavior (side-effect) triggered after the Flux terminates for any reason, including cancellation.